### DISCUSSION TITLE
OrchardCore with on premises Active Directory

### DISCUSSION THREAD

**Discussion by juergen-de** [OrchardCMS/OrchardCore#10239] [Q&A]
Hello,
does somebody use OC in an intranet environment with Active Directory authentification and authorization?
If yes, I would be happy to know how this could be done.

Thanks in advance

**Comment by hishamco** [comment]
It's one of the planned module in [OCC](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules) if you would like to contribute I will be thankful, otherwise wait for its implementation

**Comment by tropcicstefan** [comment]
In my project I customized login view and Account controller in separate module that depends on OrchardCore.Users, so it's registered afterwards. In controller I changed the logic to verify successful login with the help of novell.directory.ldap.netstandard instead of default `SignInManager<IUser>`. AD roles are mapped to orchard roles, and with every login user roles are rewritten with mapped orchard roles, so it's 'continuously up to date'.

**Reply by hishamco** [reply]
Regarding the authentication, simply you can query LDAP users, but in terms of roles you may clone the AD roles to Orchard Core roles. In both cases all the details depends on your requirements

Again if you are still interested on this we can create a module for it in OCC repo

**Comment by juergen-de** [comment]
I would be interested in that module and would help but never even looked at how modules are created. 

What would be nice in advance is to create a list of what exactly we want to achieve. For example I would definitely look for an automatic login if a users mapped OC role - AD group is matched with any of his AD groups, and not go for a customized login view. 
Where and how do we gather our requirements ?

**Reply by hishamco** [reply]
I suggest to move the discussion there, also we can list all the needed features
